Client/iOS/TodoList/Network/SessionCommonAttributes.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  SessionCommonAttributes.swift
+//  TodoList
+//
+//  Created by 백상휘 on 2022/04/05.
+//
+
+import Foundation
+
+/// 세션을 생성하는 데 필요한 기본 프로퍼티들을 저장하고 관리합니다.
+///
+/// - request: dataTask 요청할 때 필요한 구조체입니다.
+/// - httpMethod: 요청하는 httpMethod. 변경하면 자동으로 request가 업데이트 됩니다.
+/// - mimeType: 요청하는 mimeType. 변경하면 자동으로 request가 업데이트 됩니다.
+class SessionCommonAttributes {
+    
+    // MARK: - Local Properties
+    private var request: URLRequest!
+    
+    var urlString: String
+    
+    var httpMethod: HTTPMethod = .POST
+    {
+        didSet {
+            request.httpMethod = httpMethod.rawValue
+        }
+    }
+    var mimeType: MIMEType = .applicationJSON
+    {
+        didSet {
+            request.setValue(MIMEType.applicationJSON.rawValue, forHTTPHeaderField: httpHeaderField.rawValue)
+        }
+    }
+    var httpHeaderField: ReservedHTTPHeader = .contentLength
+    {
+        didSet {
+            request.setValue(MIMEType.applicationJSON.rawValue, forHTTPHeaderField: httpHeaderField.rawValue)
+        }
+    }
+    
+    private var cachePolicy = URLRequest.CachePolicy.useProtocolCachePolicy
+    private var timeoutInterval: TimeInterval = 6.0
+    
+    // MARK: - Initializers
+    init?(as string: String)
+    {
+        self.urlString = string
+        if let url = URL(string: string) {
+            reloadRequestURL(url)
+        } else {
+            return nil
+        }
+    }
+    
+    func getRequestHandler(url: URL, completionHandler: @escaping (URLRequest) -> Void) {
+        reloadRequestURL(url)
+        completionHandler(request)
+    }
+    
+    func getCurrentRequestHandler(completionHandler: @escaping (URLRequest) -> Void) {
+        completionHandler(request)
+    }
+    
+    private func reloadRequestURL(_ url: URL) {
+        request = URLRequest(url: url, cachePolicy: cachePolicy, timeoutInterval: timeoutInterval)
+        request.httpMethod = httpMethod.rawValue
+        request.setValue(mimeType.rawValue, forHTTPHeaderField: httpHeaderField.rawValue)
+    }
+    
+    // MARK: - HTTPMethod Types
+    
+    enum HTTPMethod: String
+    {
+        /// 엔티티 제출 시 사용. 서버 상태변화 등을 일으킵니다.
+        case POST = "POST"
+        /// 리소스 요청. 데이터를 받기만 합니다.
+        case GET = "GET"
```

Comment:
```suggestion
// ...
        case get = "GET"
        case post = "POST"
```

🔧 case GET 으로 작성하였다면 `GET` 으로 인식하기 때문에 별도로 작성하지 않아도 됩니다. 다만, 아래 코드와 동일한 스타일로 작성하는 것이 어떨까요?

Reply:
네넵 알겠습니다. 최대한 선언한 타입을 사용할 수 있도록 사용법을 좀 더 고려해보도록 하곘습니다!!